internal/testcase/ecs_pod.go

```diff
@@ -1 +1,159 @@
 package testcase
+
+import (
+	"context"
+	"testing"
+
+	"github.com/evergreen-ci/cocoa"
+	"github.com/evergreen-ci/cocoa/internal/testutil"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// ECSPodTestCase represents a test case for a cocoa.ECSPod.
+type ECSPodTestCase func(ctx context.Context, t *testing.T, v cocoa.Vault, pc cocoa.ECSPodCreator)
+
+// ECSPodTests returns common test cases that a cocoa.ECSPod should support.
+func ECSPodTests() map[string]ECSPodTestCase {
+	envVar := cocoa.NewEnvironmentVariable().SetName("name").SetValue("value")
+	secret := cocoa.NewEnvironmentVariable().SetName("secret1").
+		SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name1")).SetValue("value1"))
+	secretOwned := cocoa.NewEnvironmentVariable().SetName("secret2").
+		SetSecretOptions(*cocoa.NewSecretOptions().SetName(testutil.NewSecretName("name2")).SetValue("value2"))
+	secretOwned.SecretOpts.SetOwned(true)
+
+	containerDef := cocoa.NewECSContainerDefinition().SetImage("image").
+		SetEnvironmentVariables([]cocoa.EnvironmentVariable{*envVar}).
+		SetName("container").
```

Comment:
I would omit this if it's not required and not relevant to the test .

Reply:
This is required when registering the task definition--should I keep it as it is (`SetName`) or run `containerDef.Validate()` to have the validation function generate a name for the container? My hesitancy with the latter is that we would have to check the error, but there wouldn't be one. Another thing I could do is set the name with `utility.RandomString()`.

Reply:
If it's a required parameter, then it's okay to leave it.